collect_app/src/main/java/org/odk/collect/android/formentry/ODKView.java

```diff
@@ -144,7 +145,8 @@
      * @param groups          the group hierarchy that this question or field list is in
      * @param advancingPage   whether this view is being created after a forward swipe through the
      */
-    public ODKView(ComponentActivity context, final FormEntryPrompt[] questionPrompts, FormEntryCaption[] groups, boolean advancingPage, QuestionMediaManager questionMediaManager, WaitingForDataRegistry waitingForDataRegistry, AudioPlayer audioPlayer, AudioRecorderViewModel audioRecorderViewModel, FormEntryViewModel formEntryViewModel) {
+    @SuppressWarnings("PMD.ExcessiveParameterList")
```

Comment:
Since we seem to be going all in with passing in dependencies via constructors for classes that aren't Android classes, should we suppress this warning in our top-level config?

Reply:
I'm in two minds about this. When I've added them in I've generally ended up being able to get rid of it somehow, and it's felt like a good piece of pressure. That said you're right that it's not really an actual problem to have a large constructor when the alternative is often grabbing things statically. Maybe let's leave this for now and if it comes up again we revise?

Reply:
Sounds good!